src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMetricsProperties.java

```diff
@@ -35,50 +42,80 @@
 @ConfigurationProperties(prefix = "rhsm-subscriptions.metering.prometheus.metric")
 public class PrometheusMetricsProperties {
 
-  public static final String OPENSHIFT_PRODUCT_PROFILE_ID = "OpenShift";
+  @Autowired TagProfile tagProfile;
+
+  private Map<String, String> queryTemplates = new HashMap<>();
+
+  private Map<String, String> accountQueryTemplates = new HashMap<>();
+
+  /**
+   * SPEL templates do not support nested expressions so the QueryBuilder will apply template
+   * parameters a set number of times to prevent recursion.
+   */
+  private int templateParameterDepth = 3;
 
   private MetricProperties openshift = new MetricProperties();
 
-  // ENT-3835 will change the data structures used here and should refactor this method as needed
-  public Map<Uom, MetricProperties> getSupportedMetricsForProduct(String productProfileId) {
-    if (!productProfileId.equals(OPENSHIFT_PRODUCT_PROFILE_ID)) {
+  public Map<Uom, MetricProperties> getSupportedMetricsForProduct(String productTag) {
+    if (!tagProfile.tagIsPrometheusEnabled(productTag)) {
       throw new UnsupportedOperationException(
-          "Only OpenShift product profile ID is currently supported");
+          String.format("Metrics gathering for %s is not currently supported!", productTag));
     }
-    return Map.of(Uom.CORES, openshift);
-  }
 
-  // ENT-3835 will change the data structures used here and should refactor this method as needed
-  public Collection<String> getMetricsEnabledProductProfiles() {
-    return List.of(OPENSHIFT_PRODUCT_PROFILE_ID);
+    Map<Uom, MetricProperties> metrics = new EnumMap<>(Uom.class);
+    tagProfile.measurementsByTag(productTag).forEach(metric -> metrics.put(metric, openshift));
+    return metrics;
   }
```

Comment:
I'd rather see this method in TagProfile

Reply:
I agree, but I think we should wait to see what comes out of the MetricProperties/PrometheusMeteringProperties refactor in ENT-4005. I really don't like bringing MetricProperties into TagProfile.